src/initialize.cpp

```diff
@@ -54,6 +59,34 @@ int openmc_init(int argc, char* argv[], const void* intracomm)
   int err = parse_command_line(argc, argv);
   if (err) return err;
 
+#ifdef LIBMESH
+
+#ifdef _OPENMP
+  int n_threads = omp_get_max_threads();
+#else
+  int n_threads = 1;
+#endif
+
+// initialize libMesh if it hasn't been initialized already
+// (if initialized externally, the LMI object needs to be provided also)
+if (!settings::LMI && !libMesh::initialized())
+{
+#ifdef OPENMC_MPI
+  // pass command line args, empty MPI communicator, and number of threads.
+  // Because libMesh was not initialized, we assume that OpenMC is the primary
+  // application and that its main MPI comm should be used.
+  settings::LMI = std::make_unique<libMesh::LibMeshInit>(argc, argv, comm, n_threads);
+#else
+  // pass command line args, empty MPI communicator, and number of threads
+  settings::LMI = std::make_unique<libMesh::LibMeshInit>(argc, argv, 0, n_threads);
+#endif
+}
+
+if (!settings::LMI) {
+  fatal_error("libMesh::LibMeshInit object isn't set.");
+}
+#endif
+
```

Comment:
This is fine as-is, but if I had to be really nit-picky, I would maybe move this after `read_input_xml()`? Because we don't need to do these steps unless the user is actually using an unstructured mesh, right? But for simplicity's sake I'm totally fine with always doing this.

Reply:
Totally valid point. We should only need that object to be set if we're actually going to be using the capability. I'll move that check into the `LibMesh::initialize` method just before we use it to construct a mesh.

Reply:
I like that solution better than what I proposed. Would you be able to access `argv` and `argc` though?

Reply:
Oh I just meant we'd move the check that the object is set.

```
if (!settings::LMI) {
    fatal_error("Attempting to use an unstructured mesh without a libMesh::LibMeshInit object.");
}
```

Reply:
oh gotcha! yes that sounds good to me